docs/runtime/runtime-metadata.md

```diff
@@ -0,0 +1,149 @@
+                        "AccountId"
+                    ],
+                    "documentation":[
+                        " The sudoer just switched identity; the old key is supplied."
+                    ]
+                }
+            ]
+        }
+    ]
+}
+```
+
+As you can see, the metadata of overall runtime is an array of module's metadata. The included modules contain all the dependent SRML and custom runtime modules in your blockchain. 
+
+Now let's walk through each field in module's metadata:
+
+* `name`: the module's name.
+* `prefix`: the name of module's store.
+* `storage`: all the available storage entries in one module, each storage entry is then composed by:
+  * `name`: the storage entry's name.
+  * `modifier`: whether the entry is `Optional` or `Default`.
+  * `type`: the type of the storage entry, could be `PlainType`, `MapType` or `DoubleMapType`.
+  * `documentation`: notes for the entry.
+* `calls`: an array of dispatchable functions:
+  * `name`: the function's name.
+  * `args`: the arguments in function's definition, each includes parameter name and type.
+  * `documentation`: notes for the function.
+* `event`: all the defined events in a module:
+  * `name`: the event's name.
+  * `args`: the arguments in the event definition.
+  * `documentation`: notes for the event.
+  
+## Fetch The Metadata
+
+If you are using **Javascript**,  [polkadot-js/api](https://polkadot.js.org/api/) already provides friendly APIs to interact with Substrate blockchain, includes the [getMetadata](https://polkadot.js.org/api/METHODS_RPC.html#json-rpc) function.
+You can try following code snippets to fetch the metadata in this [Substrate UI](https://polkadot.js.org/apps/#/js) page:
+
+```javascript
+const { magicNumber, metadata } = await api.rpc.state.getMetadata();
+
+console.log( 'Magic number: ' + magicNumber );
+console.log( 'Metadata: ' + metadata.raw );
+```
+
+If you are using **other languages** to build your client, you can easily send **WebSocket message** or **HTTP POST request** to Substrate node endpoint by using any existing client. The message or body for `getMetadata` is:
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "state_getMetadata",
+  "params": []
+}
+```
+
+The response looks like following content, but with much more information in `result` field:
+
+```json
+{
+    "jsonrpc": "2.0",
+    "result": "0x6d6574610324187379737......44964244163636f756e74496400",
+    "id": 1
+}
+```
+
+The hexadecimal string in `result` field wraps the runtime metadata in SCALE format. Go to [Low-level Data Format](overview/low-level-data-format.md) page to learn how to decode the value and get more information about the specification and implementations.
+
+> Notes: The encoded hex-string also wraps the information of metadata version and the hard coded metadata identifier. Check the [code](https://github.com/paritytech/substrate/blob/v1.0/srml/metadata/src/lib.rs#L321) if you are interested.
```

Comment:
Should probably not be pointing to specific lines of code, even on v1.0 as some changes get backported.

Reply:
I don't know if it belongs here, but we have the hex-encoded metadata and the decoded metadata in the JS API repository, if that's of interest as a reference link:

For Substrate 1.0
hex-encoded: https://github.com/polkadot-js/api/blob/master/packages/types/src/Metadata/v4/static.ts

decoded as JSON: https://github.com/polkadot-js/api/blob/master/packages/types/src/Metadata/v4/latest.substrate.v4.json

Reply:
The decoded version is useful and been added. But the hex is not that useful to be included IMO.